paddle/framework/attr_helper.h

```diff
@@ -0,0 +1,192 @@
+#pragma once
```

Comment:
I don't quite understand what goal all the code here is trying to achieve. It might be easier if there is a design doc like [paddle/framework/variable.md](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/variable.md) for [paddle/framework/variable.h](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/variable.h).